### ISSUE TITLE
EOS get_vlans doesn't show missing VLANs

### ISSUE THREAD

**Issue by aj-cruz** [networktocode/network-importer#285]
### Environment
* Python version:  3.9.15
* Network Importer version:  3.1.0
* Arista vEOS-Lab 4.29.0.2F

### Steps to Reproduce
1. Configure an EOS device with a management interface, enable eAPI, and configure a couple interfaces and VLANs
```
interface Management1
   ip address 192.168.254.56/24
!
management api http-commands
   no shutdown
!
interface Ethernet1
   no switchport
   ip address 10.201.0.1/24
!
interface Ethernet2
   switchport mode trunk
!
vlan 29
   name Desktops
!
vlan 30
   name Phones
```
2. Onboard EOS device into Nautobot, confirm device exists in Nautobot with management interface and primary IP
3. Run ```network-importer check --update-configs --limit="site=tdc"```

### Expected Behavior
Network Importer reports two interfaces and two vlans missing in Nautobot

### Observed Behavior
Network Importer reports two interfaces but no vlans missing in Nautobot:
```
root@b889f97b554e:/source/network_importer# network-importer check --update-configs --limit="site=tdc"
2022-11-10 03:17:34,637 - network-importer - INFO - Updating configuration from devices .. 
2022-11-10 03:17:34,895 - network-importer - INFO - TUL-CORE1 | Configuration file updated
2022-11-10 03:17:34,896 - network-importer - INFO - Import SOT Model
2022-11-10 03:17:35,635 - network-importer - INFO - Import Network Model
2022-11-10 03:17:41,541 - network-importer - INFO - Collecting vlans information from devices .. 
device
  device: TUL-CORE1
    interface
      interface: Ethernet1 MISSING in Nautobot
        ip_address
          ip_address: TUL-CORE1__Ethernet1__10.201.0.1/24 MISSING in Nautobot
      interface: Ethernet2 MISSING in Nautobot
2022-11-10 03:17:41,615 - network-importer - INFO - Execution finished, processed 1 device(s)
```

EDIT: I enabled debugging and got this:
```
2022-11-10 05:26:02,057 - network-importer - INFO - Collecting vlans information from devices .. 
2022-11-10 05:26:02,058 - network-importer - DEBUG - Executing dispatcher for TUL-CORE1 (arista_eos)
2022-11-10 05:26:02,059 - network-importer - DEBUG - Found driver network_importer.drivers.arista_eos
2022-11-10 05:26:02,114 - nornir.core.task - ERROR - Host 'TUL-CORE1': task 'get_vlans' failed with traceback:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/local/network_importer/drivers/arista_eos.py", line 50, in get_vlans
    results.vlans.append(Vlan(name=data["name"], id=vid))
AttributeError: 'list' object has no attribute 'vlans'

2022-11-10 05:26:02,115 - nornir.core.task - ERROR - Host 'TUL-CORE1': task 'dispatcher' failed with traceback:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/local/network_importer/drivers/__init__.py", line 46, in dispatcher
    result = task.run(task=driver_task)
  File "/usr/local/lib/python3.9/site-packages/nornir/core/task.py", line 174, in run
    raise NornirSubTaskError(task=run_task, result=r)
nornir.core.exceptions.NornirSubTaskError: Subtask: get_vlans (failed)
```

**Comment by aj-cruz** [comment]
I changed my network_importer.toml from ```import_vlans = "cli"``` to ```import_vlans = true```
Now it thinks every possible vlan is missing from Nautobot:
```
---OUTPUT OMITTED FOR BREVITY---
      vlan: tdc__3273 MISSING in Nautobot
      vlan: tdc__3274 MISSING in Nautobot
      vlan: tdc__3275 MISSING in Nautobot
      vlan: tdc__3276 MISSING in Nautobot
      vlan: tdc__3261 MISSING in Nautobot
      vlan: tdc__3262 MISSING in Nautobot
      vlan: tdc__3263 MISSING in Nautobot
      vlan: tdc__3248 MISSING in Nautobot
      vlan: tdc__3249 MISSING in Nautobot
      vlan: tdc__3250 MISSING in Nautobot
      vlan: tdc__3251 MISSING in Nautobot
      vlan: tdc__3252 MISSING in Nautobot
device
  device: TUL-CORE1
    interface
      interface: Ethernet1 MISSING in Nautobot
        ip_address
          ip_address: TUL-CORE1__Ethernet1__10.201.0.1/24 MISSING in Nautobot
      interface: Ethernet2 MISSING in Nautobot
2022-11-10 03:33:14,134 - network-importer - INFO - Execution finished, processed 1 device(s)
```

**Comment by aj-cruz** [comment]
I did some digging in /local/network_importer/drivers/arista_eos.py
A few things I noticed regarding this part of the code:
```
        results = Vlans()

        nr_device = task.host.get_connection("napalm", task.nornir.config)
        eos_device = nr_device.device
        results = eos_device.run_commands(["show vlan"])
        
        if not isinstance(results[0], dict) or "vlans" not in results[0]:
            LOGGER.warning("%s | No vlans information returned", task.host.name)
            return Result(host=task.host, result=False)

        for vid, data in results[0]["vlans"].items():
            results.vlans.append(Vlan(name=data["name"], id=vid))
```
1. It looks like ```results = Vlans()``` is being overwritten by ```results = eos_device.run_commands(["show vlan"])```
2. The pydantic validator is being passed the parameter ```id``` but it looks like the validator requires ```vid```
3. It appears we need to return a dict, but we're returning the raw results instead.

**Comment by aj-cruz** [comment]
I fixed it on my local install by changing arista_eos.py to this:
```
        results = Vlans()

        nr_device = task.host.get_connection("napalm", task.nornir.config)
        eos_device = nr_device.device
        device_results = eos_device.run_commands(["show vlan"])

        if not isinstance(device_results[0], dict) or "vlans" not in device_results[0]:
            LOGGER.warning("%s | No vlans information returned", task.host.name)
            return Result(host=task.host, result=False)

        for vid, data in device_results[0]["vlans"].items():
            results.vlans.append(Vlan(name=data["name"], vid=int(vid)))

        return Result(host=task.host, result=results.dict())
```

**Comment by aj-cruz** [comment]
Pull Request for proposed fix: [https://github.com/networktocode/network-importer/pull/286](https://github.com/networktocode/network-importer/pull/286)

**Comment by jvanderaa** [comment]
Closed via #268.
